doc/repository-spec-v2.md

```diff
@@ -10,7 +9,26 @@ This can be overwritten in the `dependencies` of a `pubspec.yaml` or specified
 with the environment variable `PUB_HOSTED_URL`. In the rest of this document we
 shall refer to the base URL as `PUB_HOSTED_URL`.
 
+## Authentication
+
+All relevant commands will add an `Authorization` header with `Bearer <token>` value if
```

Comment:
A repository specification does not speak about "commands", it speaks about the interface between _a hosted pub package repository_ and the _pub client_.

This stuff belongs in our documentation on `dart.dev`, but not in the specification for the repository API.

-----------

What we should state here is how a server signals that authorization is required, and how a client is supposed to provide said authorization.

Furthermore we should clearly state which API end-points require authorization.

Should the `archive_url` returned from `GET <PUB_HOSTED_URL>/api/packages/<PACKAGE>` require authorization. My initial inclination is no it should not. The 3rd party server can simply put a _signed URL_ in the response (though I can see that leaking in to logs).

We use this because we actually have packages hosted on a different hostname, we use a GCS storage bucket to serve packages (which is nice).

Also we should determine if authorization may be provided when following redirects. I think it's only the `archive_url` which is says it may have redirects. The others that use redirects are deprecated at this point (so the client will never use them, they are only documented because servers supporting older clients should implement them).

Reply:
Ok. I will revert the commit for this file until we know for sure what to include and what not.

I am not completely sure that the `archive_url` can be an unauthenticated url; I think hosts like `GitHub Packages` would require an authenticated url for the real package file as otherwise that url can become exposed. A signed URL could be a solution, but then it needs to be different for every get.

An alternative option is that we support a specific header to be returned from that call that gives an indication if we need to forward the authentication header or not (default=true).